File: src/app/frontend/src/api/push.ts
import request from './request'

export interface SmtpConfig {
  host: string
  port: number
  use_tls: boolean
  use_ssl: boolean
  username: string
  from_email: string
  from_name: string
  password: string // 脱敏展示值
}

export interface PushRun {
  id: number
  rule_id: number
  trigger_mode: string
  recipients: string[]
  event_count: number
  status: 'succeeded' | 'failed' | 'no_new'
  error: string | null
  // push-email-preview-inline-figures：邮件内容留存字段
  subject: string | null
  attachment_summary: Array<Record<string, unknown>> | null
  has_preview: boolean
  started_at: string | null
  finished_at: string | null
  created_at: string
}

export interface PushRunPreview {
  subject: string | null
  html: string | null
  attachments: Array<Record<string, unknown>> | null
}

// ---- SMTP config（全局邮件通道，归属「任务分析」页）----
export const getSmtpApi = () => request.get<unknown, SmtpConfig>('/api/push/smtp')

export const putSmtpApi = (data: {
  host: string
  port: number
  use_tls: boolean
  use_ssl: boolean
  username: string
  password: string // 空表示保留旧密码
  from_email: string
  from_name: string
}) => request.put<unknown, SmtpConfig>('/api/push/smtp', data)

export const testSmtpApi = (to_email: string) =>
  request.post<unknown, { ok: boolean; error: string | null }>('/api/push/smtp/test', { to_email })

// push-email-preview-inline-figures：推送历史邮件预览
export const getPushRunPreviewApi = (taskId: number, runId: number) =>
  request.get<unknown, PushRunPreview>(`/api/analysis-tasks/${taskId}/push/runs/${runId}/preview`)

File: src/app/backend/schemas/push.py
"""Push schemas: SMTP config, push rules, push runs (history)."""
from __future__ import annotations

from pydantic import BaseModel, model_validator

from .common import BeijingDatetime, ORMBase


class SmtpConfigOut(BaseModel):
    host: str
    port: int
    use_tls: bool
    use_ssl: bool
    username: str
    from_email: str
    from_name: str
    password: str  # 脱敏后的展示值


class SmtpConfigIn(BaseModel):
    host: str = ""
    port: int = 25
    use_tls: bool = False
    use_ssl: bool = False
    username: str = ""
    password: str = ""  # 空表示保留旧密码（避免前端回传脱敏值覆盖）
    from_email: str = ""
    from_name: str = "信息智能分析系统"


class TestEmailRequest(BaseModel):
    to_email: str


class PushRunOut(ORMBase):
    id: int
    rule_id: int
    trigger_mode: str
    recipients: list[str]
    event_count: int
    status: str
    error: str | None
    # push-email-preview-inline-figures：邮件内容留存字段
    subject: str | None = None
    attachment_summary: list | None = None
    has_preview: bool = False
    started_at: BeijingDatetime | None
    finished_at: BeijingDatetime | None
    created_at: BeijingDatetime


class PushRunPreviewOut(BaseModel):
    """推送历史预览接口返回（push-email-preview-inline-figures）。"""

    subject: str | None = None
    html: str | None = None
    attachments: list | None = None

